src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java

```diff
@@ -230,6 +230,15 @@ private static String getDomainNameFromURL(String url) {
      *                                  navigation
      */
     public static void navigateToURL(WebDriver driver, String targetUrl, String targetUrlAfterRedirection) {
+        // check if the user wrote the URL ended with "/"
+        String finalUrl = null;
+        if (System.getProperty("baseURL").endsWith("/")) {
+            finalUrl = System.getProperty("baseURL").substring(0, System.getProperty("baseURL").length() - 1);
+        }
+        // check if the user sends the URL with the abbreviation of "./"
+        if (targetUrl.startsWith(".")) {
+            targetUrl = targetUrl.replaceFirst(".", finalUrl);
```

Comment:
should this be "finalUrl =" ??

Reply:
I changed it to Modified URL and added cases to handle If user selects to navigate with the new feature but didn't add a base Url